webapp/client/src/components/FormFieldYesNo.js

```diff
@@ -0,0 +1,162 @@
+  }
+
+  & .switch-yes:hover,
+  .switch-no:hover,
+  .switch-yes.active:hover,
+  .switch-no.active:hover,
+  .switch-yes:focus-within:hover,
+  .switch-no:focus-within:hover {
+    color: var(--inverse-text-color) !important;
+
+    background: var(--link-hover-color) !important;
+    border: 1px solid var(--link-hover-color) !important;
+  }
+
+  & input[type="radio"] {
+    opacity: 0;
+    position: absolute;
+  }
+`;
+
+function FormFieldYesNo({
+  fieldName,
+  fieldId,
+  value,
+  required,
+  autofocus,
+  label,
+  details,
+  errors,
+  onChangeHandler,
+}) {
+  const [selectedValue, setSelectedValue] = useState(value);
+  const yesFieldId = `${fieldId}-yes`;
+  const noFieldId = `${fieldId}-no`;
+
+  const toggleYesNoButton = (event) => {
+    setSelectedValue(event.target.value);
+  };
+
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+        cols: "6",
+      }}
+      hideLabel
+      render={() => (
+        <YesNo>
+          <FieldLabelForSeparatedFields {...{ label, fieldId, details }} />
+          <fieldset id={fieldId} name={fieldId}>
+            <label
+              htmlFor={yesFieldId}
+              className={`switch-yes ${
+                selectedValue === "yes" ? "active" : ""
+              }`}
+            >
+              <input
+                type="radio"
+                id={yesFieldId}
+                name={fieldId}
+                required={required}
+                value="yes"
+                active={selectedValue === "yes"}
+                onClick={toggleYesNoButton}
+                onChange={onChangeHandler}
+              />
+              Ja
+            </label>
+            <label
+              htmlFor={noFieldId}
+              className={`switch-no ${selectedValue === "no" ? "active" : ""}`}
+            >
+              <input
+                type="radio"
+                id={noFieldId}
+                name={fieldId}
+                required={required}
+                autoFocus={autofocus || Boolean(errors.length)}
+                value="no"
+                active={selectedValue === "no"}
+                onClick={toggleYesNoButton}
+                onChange={onChangeHandler}
+              />
+              Nein
+            </label>
+          </fieldset>
+        </YesNo>
+      )}
+    />
+  );
+}
+
+FormFieldYesNo.propTypes = {
+  fieldName: PropTypes.string.isRequired,
+  fieldId: PropTypes.string.isRequired,
+  value: PropTypes.string,
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+  details: FieldLabelForSeparatedFields.propTypes.details,
+  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
+  required: PropTypes.bool,
+  autofocus: PropTypes.bool,
+  onChangeHandler: PropTypes.func,
+};
+
+FormFieldYesNo.defaultProps = {
+  value: undefined,
+  required: false,
+  autofocus: false,
+  details: FieldLabelForSeparatedFields.defaultProps.details,
+  onChangeHandler: () => {},
```

Comment:
There's a code smell flagged for this… First I didn't agree, but thinking it over I now agree it'd be better to default this to `undefined`. The code should work without any other modification and it's clearer this way that nothing has been set for this.

Reply:
I agree. At some point I had a problem with using `undefined` as a defaultProps for a function. That's why I did it like this because this was recommended in However, as I can not reproduce this I have changed it to `undefined` 👍